Pokemon Unity/Assets/Scripts2/Pokemon/Pokemon.cs

```diff
@@ -269,6 +265,83 @@ public Pokemon(Pokemons pokemon) : this()
 
             //calcStats();
         }
+
+        public Pokemon(string nickName, int form,
+            Pokemons species, Abilities ability,
+            Nature nature,
+            bool isShiny, bool? gender,
+            int[] pokerus, int pokerusStrain,
+            int? shadowLevel,
+            int currentHp, Items item,
+            byte[] iv, byte[] ev, 
+            int obtainedLevel, int currentLevel, int currentExp,
+            int happiness, Status status, int statusCount,
+            int eggSteps, Items ballUsed,
+            string mail, Move[] moves,
+            Ribbon[] ribbons, bool[] markings,
+            int personalId,
+            ObtainedMethod obtainedMethod,
+            DateTimeOffset timeReceived, DateTimeOffset timeEggHatched) : this(species)
+        {
+            //Check to see if nickName is filled
+            if (nickName != null || nickName != string.Empty)
+            {
+                name = nickName;
+            }
+            else
+            {
+                name = null;
+            }
+
+            Form = form;
+            _base = Pokemon.PokemonData.GetPokemon(species);
```

Comment:
i personally would just comment this line out, because line 284 `: this(species)` does exactly the same thing here as line 297. But it's not a big deal, just saying it's redundant in use, as it's purpose has already be established, and we're doing it again. (Doesnt harm code, just speaking more on what it is we're looking at)